en/upgrade-tool.rst

```diff
@@ -0,0 +1,107 @@
+Upgrade Tool
+############
+
+Upgrading to CakePHP 3 from CakePHP 2.x requires several mechanical
+transformations, such as adding namespaces. To assist in making these mechanical
+changes easier, CakePHP provides a CLI based upgrade tool.
+
+Installation
+============
+
+The upgrade tool is installed as a standalone application. You'll need to clone
+the upgrade tool with git, and install the dependencies with composer::
+
+    git clone https://github.com/cakephp/upgrade.git
+    cd upgrade
+    php ../composer.phar install
+
+At this point you should be able to get the help for the upgrade tool::
+
+    cd upgrade
+    bin/cake upgrade --help
+
+The above should output something like the following::
+
+    Welcome to CakePHP v3.0.8 Console
+    ---------------------------------------------------------------
+    App : src
+    Path: /Users/markstory/Sites/cake_plugins/upgrade/src/
+    ---------------------------------------------------------------
+    A shell to help automate upgrading from CakePHP 2.x to 3.x. Be sure to
+    have a backup of your application before running these commands.
+
+    Usage:
+    cake upgrade [subcommand] [-h] [-v] [-q]
+
+    Subcommands:
+
+    locations           Move files/directories around. Run this *before*
```

Comment:
Not sure if we want to show al that, because it could become out of date/we need to keep it updated.
This could lead to confusion etc...

I think it's enough to explain how to get the list of possible commands https://github.com/cakephp/docs/pull/2929/files#diff-a35180b2b47fe158f12f141ab9583869R21